src/controllers/movie.js

```diff
@@ -0,0 +1,159 @@
+const {
+  PrismaClientKnownRequestError,
+  PrismaClientValidationError,
+} = require("@prisma/client/runtime/library");
+const {
+  getMoviesDb,
+  getMoviesWhereAndDb,
+  getMoviesWhereLtDb,
+  getMoviesWhereGtDb,
+  createMovieDb,
+  createMovieWithScreeningsDb,
+  getMovieByIdDb,
+  updateMovieDb,
+  getMovieByTitleDb,
+  updateMovieWithScreeningsDb,
+} = require("../domains/movie");
+
+const getMovies = async (req, res) => {
+  let movies;
+  const { runtimeLt, runtimeGt } = req.query;
+  if (runtimeLt && runtimeGt) {
+    const runtimeLtNum = Number(runtimeLt);
+    const runtimeGtNum = Number(runtimeGt);
+    movies = await getMoviesWhereAndDb(runtimeGtNum, runtimeLtNum);
+  }
+
+  if (runtimeLt && !runtimeGt) {
+    const runtimeLtNum = Number(runtimeLt);
+    movies = await getMoviesWhereLtDb(runtimeLtNum);
+  }
+
+  if (runtimeGt && !runtimeLt) {
+    const runtimeGtNum = Number(runtimeGt);
+    movies = await getMoviesWhereGtDb(runtimeGtNum);
+  }
+
+  if (!runtimeLt && !runtimeGt) {
+    movies = await getMoviesDb();
+  }
+
+  res.json({ movies: movies });
+};
+
+const createMovie = async (req, res) => {
+  const { title, runtimeMins, screenings } = req.body;
+  let createdMovie;
+
+  if (!screenings || screenings.length === 0) {
+    try {
+      createdMovie = await createMovieDb(title, runtimeMins);
+    } catch (e) {
+      if (e instanceof PrismaClientKnownRequestError) {
+        if (e.code === "P2002") {
+          res
+            .status(409)
+            .json({ error: "A movie with the provided title already exists" });
+          return;
+        }
+      }
+      if (e instanceof PrismaClientValidationError) {
+        // e.code is undefined for P1012 ("Arguement {} is missing.")
+        // hence testing that the ends of the message matches what
+        // we'd expect instead.
+        if (e.message.endsWith("is missing.")) {
+          res.status(400).json({ error: "Missing fiels in request body" });
+          return;
+        }
+      }
+    }
+  }
+
+  if (screenings && screenings.length !== 0) {
```

Comment:
`if (screenings && screenings.length)`

Reply:
is that because 0 would evaluate to false, and any other numerical value to true, making the whole `!== 0` bit redundant?